code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -40,102 +52,276 @@
-		owner.visible_message("<span class='danger'>[src] blocks [attack_text], sending out jets of flame!</span>")
-		playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
-		for(var/mob/living/carbon/C in range(6, owner))
-			if(C != owner)
-				C.adjust_fire_stacks(8)
-				C.IgniteMob()
-		owner.set_fire_stacks(-20)
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
+	if(!owner.equip_to_slot_if_possible(gloves, ITEM_SLOT_GLOVES, bypass_equip_delay_self = TRUE))
+		to_chat(owner, "<span class='warning'>Something is blocking the machine from equipping the reactive incendiary gloves on you!</span>")
+		return FALSE
+	owner.visible_message("<span class='notice'>[src] extends some metal gauntlets and they lock around [owner]'s hands!</span>")
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/fire/deactivate_effect(mob/living/carbon/human/owner)
+	if(gloves)
+		owner.visible_message("<span class='notice'>[src] disengages [gloves] from you.</span>")
+		user.transferItemToLoc(O, src, force = TRUE) //force allows it to move despite being nodrop
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/fire/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.drop_all_held_items()
+	var/obj/item/gun/ballistic/rifle/boltaction/enchanted/arcane_barrage/flame_jets/jets = new
+	if(!owner.put_in_hands(jets))
+		owner.visible_message("<span class='danger'>[src] attempts to react, but the flame jets are blocked by what [owner] is holding!</span>")
+		return FALSE
+	owner.visible_message("<span class='danger'>[src] blocks [attack_text], opening the flame jets!</span>")
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/fire/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>[src] just makes [attack_text] worse by jamming and rupturing, releasing molten death on [owner]!</span>")
+	playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
+	owner.adjust_fire_stacks(12)
+	owner.IgniteMob()
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
 	return FALSE
 
+/obj/item/clothing/gloves/flame_jets
+	name = "reactive incendiary gauntlets"
+	desc = "A pair of metal gauntlets that are hooked up to the reactive incendiary armor. When threatened, you will be able to spew flame from them... In theory!"
+	icon_state = "flamejet"
+	inhand_icon_state = "blackgloves"
+	permeability_coefficient = 0.9
+	heat_protection = HANDS
+	max_heat_protection_temperature = GLOVES_MAX_TEMP_PROTECT
+	resistance_flags = NONE
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 70, ACID = 30)
+
+/obj/item/clothing/gloves/flame_jets/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, "flame jets")
```

Comment:
what if you pop your own arm off using autonomy (and then reattach it using ligament hooks)?

Reply:
see above